test/DeployCampaign.integration.test.ts

```diff
@@ -0,0 +1,315 @@
+import { getConfig } from "../src/deploy/campaign/environments";
+import { getLogger } from "../src/deploy/logger/create-logger";
+import { runZnsCampaign } from "../src/deploy/zns-campaign";
+import { IDeployCampaignConfig, TZNSContractState } from "../src/deploy/campaign/types";
+import { ethers } from "ethers";
+import { IDistributionConfig } from "./helpers/types";
+import { expect } from "chai";
+import { hashDomainLabel } from "./helpers";
+import {
+  approveBulk,
+  getPriceBulk,
+  mintBulk,
+  registerRootDomainBulk,
+  registerSubdomainBulk,
+} from "./helpers/deploy-helpers";
+
+describe("DeployCampaign - Integration", () => {
+  let deployer : SignerWithAddress;
+  let zeroVault : SignerWithAddress;
+  let domainAddressMock : SignerWithAddress;
+
+  // Minters
+  let userA : SignerWithAddress;
+  let userB : SignerWithAddress;
+  let userC : SignerWithAddress;
+  let userD : SignerWithAddress;
+  let userE : SignerWithAddress;
+  let userF : SignerWithAddress;
+
+  let zns : TZNSContractState;
+
+  let config : IDeployCampaignConfig;
+
+  let users : Array<SignerWithAddress>;
+  let distConfig : IDistributionConfig;
+
+  const logger = getLogger();
+
+  // Default baselength is 4, maxLength is 50
+  const shortDomain = "wild"; // Length 4
+  const mediumDomain = "wilder"; // Length 6
+  const longDomain = "wilderwilderwilderwilderwilderwilderwilderwilderwil"; // Length 51
+  const shortHash = hashDomainLabel(shortDomain);
+  const mediumHash = hashDomainLabel(mediumDomain);
+  const longHash = hashDomainLabel(longDomain);
+
+  const freeShortSubdomain = "subd"; // Length 4
+  const freeMediumSubdomain = "subder"; // Length 6
+  const freeLongSubdomain = "subderwilderwilderwilderwilderwilderwilderwilderwil"; // Length 51
+
+  // Resolve through async call `hashWithParent` in `before` hook
+  let freeShortSubHash : string;
+  let freeMediumSubHash : string;
+  let freeLongSubHash : string;
+
+  const paidShortSubdomain = "surf"; // Length 4
+  const paidMediumSubdomain = "surfer"; // Length 6
+  const paidLongSubdomain = "surferwilderwilderwilderwilderwilderwilderwilderwil"; // Length 51
+
+  let paidShortSubHash : string;
+  let paidMediumSubHash : string;
+  let paidLongSubHash : string;
+
+  const mintAmount = ethers.utils.parseEther("10000000");
+
+  const domains = [shortDomain, mediumDomain, longDomain];
+
+  before(async () => {
+    [deployer, zeroVault, domainAddressMock, userA, userB, userC, userD, userE, userF] = await hre.ethers.getSigners();
+
+    // Reads `ENV_LEVEL` environment variable to determine rules to be enforced
+    config = getConfig(deployer, zeroVault);
+
+    config.mockMeowToken = hre.network.name === "hardhat";
+    const campaign = await runZnsCampaign({ config, dbVersion: "1.0.0" });
+
+    zns = campaign.state.contracts;
+
+    //  CurvePricer, stake, open
+    distConfig = {
+      pricerContract: zns.curvePricer.address,
+      paymentType: 1,
+      accessType: 1,
```

Comment:
please use existing enums already created for each of these fields:

https://github.com/zer0-os/zNS/blob/feat/verify-monitor-contracts/test/helpers/constants.ts#L20-L37